test/hermes_test/commands/harvest/test_git.py

```diff
@@ -0,0 +1,125 @@
+
+import hermes.commands.harvest.git as harvest
+from hermes.model.context import HermesContext, HermesHarvestContext
+
+
+@pytest.fixture
+def harvest_ctx(request: FixtureRequest):
+    ctx = HermesContext()
+    return HermesHarvestContext(
+        ctx,
+        EntryPoint(name=request.function.__name__, group='hermes.harvest', value='hermes_test:ctx')
+    )
+
+
+@pytest.fixture
+def codemeta():
+    return json.loads("""
+{
+  "@context": [
+    "https://doi.org/10.5063/schema/codemeta-2.0",
+    {
+      "hermes": "https://software-metadata.pub/ns/hermes/"
+    }
+  ],
+  "@type": "SoftwareSourceCode",
+  "contributor": [
+    {
+      "@type": "Person",
+      "name": "Git Author + Committer",
+      "email": "e@mail.com",
+      "hermes:contributionRole": [
+        {
+          "@type": "Role",
+          "roleName": "git author",
+          "startTime": "2023-01-13T08:57:12+01:00",
+          "endTime": "2023-01-13T08:57:12+01:00"
+        },
+        {
+          "@type": "Role",
+          "roleName": "git committer",
+          "startTime": "2023-01-13T08:57:12+01:00",
+          "endTime": "2023-01-13T08:57:12+01:00"
+        }
+      ]
+    },
+    {
+      "@type": "Person",
+      "name": "Git Author",
+      "email": "f@mail.com",
+      "hermes:contributionRole": [
+        {
+          "@type": "Role",
+          "roleName": "git author",
+          "startTime": "2023-01-13T08:57:12+01:00",
+          "endTime": "2023-01-13T08:57:12+01:00"
+        }
+      ]
+    },
+    {
+      "@type": "Person",
+      "name": "Git Committer",
+      "email": "g@mail.com",
+      "hermes:contributionRole": [
+        {
+          "@type": "Role",
+          "roleName": "git committer",
+          "startTime": "2023-01-13T08:57:12+01:00",
+          "endTime": "2023-01-13T08:57:12+01:00"
+        }
+      ]
+    }
+  ]
+}
+    """)
+
+
+@pytest.fixture
+def git_log_output():
+    return '''
+Git Author + Committer|e@mail.com|2023-01-13T08:57:12+01:00|Git Author + Committer|e@mail.com|2023-01-13T08:57:12+01:00
+Git Author|f@mail.com|2023-01-13T08:57:12+01:00|Git Committer|g@mail.com|2023-01-13T08:57:12+01:00
+    '''
+
+
+@pytest.fixture
+def mock_click_ctx():
+    ctx = Mock()
+    ctx.parent = Mock(params={'path': '.'})
+    return ctx
+
+
+@patch("hermes.commands.harvest.git.subprocess.run")
+def test_harvest_git(mock_run, mock_click_ctx, harvest_ctx, git_log_output, codemeta):
+    # Mocking the output of the subprocess call to git log to retrieve author and committer info
+    mock_stdout = MagicMock()
+    mock_stdout.configure_mock(
+        **{
+            "stdout.decode.return_value": git_log_output,
+            "returncode": 0
+        }
+    )
+
+    # Mocking the output of the subprocess call to git rev-parse to retrieve branch id
+    mock_get_branch = Mock()
+    mock_get_branch.returncode = 0
+
+    # Define outputs from "running" the subprocess mock
+    mock_run.side_effect = [mock_get_branch, mock_stdout]
```

Comment:
`subprocess.run` is called two times. The mock only provides valid output for the second call. The first call
https://github.com/hermes-hmc/workflow/blob/f02ff2a2b7a9ea2b2dd278fd9c0c0b827ac41bbc/src/hermes/commands/harvest/git.py#L287
is to get the current branch name/state. If the harvester would do sanity checks on the git output, `harvest_git` would be expected to fail here. This information is also passed to the context update
https://github.com/hermes-hmc/workflow/blob/f02ff2a2b7a9ea2b2dd278fd9c0c0b827ac41bbc/src/hermes/commands/harvest/git.py#L315-L323
.. but apparently does not end up in the context json, otherwise the test would fail ... what is going on here?

Reply:
The branch is passed to `HermesHarvestContext.updat_from` as a keyword arg and remian unused therein. I.e. the proposed mock works *until* we decide to record the branch in the harvest context.

Reply:
> This information is also passed to the context update
> 
> https://github.com/hermes-hmc/workflow/blob/f02ff2a2b7a9ea2b2dd278fd9c0c0b827ac41bbc/src/hermes/commands/harvest/git.py#L315-L323
> 
> .. but apparently does not end up in the context json, otherwise the test would fail ... what is going on here?

Huh, must've slipped through review. The information *isn't* actually passed to the context. To pass it, either include it in the `dict` itself

```python
    ctx.update_from({ 
     '@context': [ 
         "https://doi.org/10.5063/schema/codemeta-2.0", 
         {'hermes': 'https://software-metadata.pub/ns/hermes/'} 
     ], 
  
     '@type': "SoftwareSourceCode", 
     'contributor': [contributor.to_codemeta() for contributor in git_contributors._all], 
     'branch': git_branch
 }) 
```

Or add it separately (which I'd prefer for reasons of readability) with `ctx.update('branch', git_branch)`.
This is a bug, tracked in